### DISCUSSION TITLE
How to add mapping starting with `s` on a directory

### DISCUSSION THREAD

**Discussion by enewbury** [nvim-neo-tree/neo-tree.nvim#1401] [Q&A] [answered]
I'm trying to add telescope mapping to grep in a directory, so I added the mapping `sd` (search directory) however, it seems that the "s" key when typed over a directory behaves like pressing <cr> and opens and closes the sub tree. Is this a normal behavior, and if so, how do I get it to wait for more keys rather than returning immediately? 
Note it also does this with the t key.

**Comment by pysan3** [comment, marked as answer]
`s` is mapped to open in a vsplit by default as you can see here: https://github.com/nvim-neo-tree/neo-tree.nvim/blob/120a83eb7c3c240d5298122111e9f65f0ee5bc8f/lua/neo-tree/defaults.lua#L374.

You can add `["s"] = "noop"` in your config to disable this behavior, or change it to
`["s"] = { "open_vsplit", nowait = false },` to keep the default behavior but wait for additional keys and not decide immediately. See `:h map-arguments` for more info about the options.

You can find what other keys are mapped to in this `defaults.lua` file as well and do the exact thing for each key.

**Reply by enewbury** [reply]
This vsplit key binding only works when your cusor is over a file. When the cursor is over a directory this behavior seems to change and simply opens and closes the subtree. Id rather not unbind since I do use that mapping, but it does work to unbind, so I'm going to use `tf` (telescope find) for now since I don't use tabs much

**Reply by pysan3** [reply]
I forgot to mention. Yes vsplit if file / toggle node if is dir. This is indeed the default behavior. 

Doesn't nowait work for you?
And tbh I map `["v"] = "open_vsplit"` which makes more sense imo.
